dttools/src/buffer.c

```diff
@@ -66,12 +66,14 @@ void buffer_free(buffer_t * b)
 	}
 }
 
-/* make room for at least n chars */
-static int grow(buffer_t * b, size_t n)
+int buffer_grow(buffer_t * b, size_t n)
 {
 	size_t used = inuse(b);
 	size_t newlen = sizeof(b->initial); /* current buf is always at least as big as b->initial */
 
+	/* Keep using the initial buffer if possible */
```

Comment:
Doesn't realloc do this anyway?

Reply:
If we're still using the initial buffer in the struct, there would be an unconditional `malloc()` [here](https://github.com/cooperative-computing-lab/cctools/blob/6c252547900f74aca35a1560c2e332c00f6d2a5c/dttools/src/buffer.c#L91). It looks like the other functions check the available space before calling `grow()`, so to make this a friendly public function I checked that here too.